x-pack/plugins/lens/public/editor_frame_service/editor_frame/suggestion_helpers.ts

```diff
@@ -221,3 +223,35 @@ export function switchToSuggestion(
 
   dispatch(action);
 }
+
+export function getTopSuggestionForField(
+  datasourceLayers: Record<string, DatasourcePublicAPI>,
+  activeVisualizationId: string | null,
+  visualizationMap: Record<string, Visualization<unknown>>,
+  visualizationState: unknown,
+  datasource: Datasource,
+  datasourceStates: Record<string, { state: unknown; isLoading: boolean }>,
+  field: Dragging
+) {
+  const hasData = Object.values(datasourceLayers).some(
+    (datasourceLayer) => datasourceLayer.getTableSpec().length > 0
+  );
+
+  const mainPalette =
+    activeVisualizationId && visualizationMap[activeVisualizationId]?.getMainPalette
+      ? visualizationMap[activeVisualizationId].getMainPalette?.(visualizationState)
+      : undefined;
+  const suggestions = getSuggestions({
+    datasourceMap: { [datasource.id]: datasource },
+    datasourceStates,
+    visualizationMap:
+      hasData && activeVisualizationId
+        ? { [activeVisualizationId]: visualizationMap[activeVisualizationId] }
+        : visualizationMap,
+    activeVisualizationId,
+    visualizationState,
+    field,
+    mainPalette,
+  });
+  return suggestions.find((s) => s.visualizationId === activeVisualizationId) || suggestions[0];
```

Comment:
I tried to break the suggestion logic here, and it does seem to be working consistently with what we have on master, but might benefit from some extra tests. It seems like we are intentionally switching the `subVisualizationId` on some chart types, like between donut -> pie when you add a second agg. Is this tested?